tests/integration/sentinel/schedules/schedules-start-from.spec.js

```diff
@@ -0,0 +1,253 @@
+
+    const clinic = {
+      _id: uuid(),
+      type: 'data_record',
+      form: 'FORM',
+      from: '+11111111',
+      fields: {
+        place_id: 'the_clinic',
+      },
+      some_date_field: startDate,
+      reported_date: moment().valueOf(),
+      contact: {
+        _id: 'middle_man',
+        parent: { _id: 'health_center', parent: { _id: 'district_hospital' } }
+      },
+    };
+
+    const expectedMessage2 = (state) => createExpectedMessage('sch1', 2, state, 'schedule from field in array');
+
+    const expectedMessage3 = (state) => createExpectedMessage('sch2', 1, state, 'schedule from reported_date');
+
+    const expectedMessage4 = (state) => createExpectedMessage('sch2', 1, state, 'schedule from reported_date');
+
+    const expectedMessage5 = (state) => createExpectedMessage('sch3', 1, state, 'schedule from first existing field');
+
+    const expectedMessage6 = (state) => createExpectedMessage('sch3', 1, state, 'another schedule from array');
+
+    return utils
+      .updateSettings(settings, 'sentinel')
+      .then(() => utils.saveDocs([patient, clinic]))
+      .then(() => sentinelUtils.waitForSentinel([patient._id, clinic._id]))
+      .then(() => sentinelUtils.getInfoDocs([patient._id, clinic._id]))
+      .then(([infoWithPatient, infoWithClinic]) => {
+        chai.expect(infoWithPatient).to.deep.nested.include({ 'transitions.registration.ok': true });
+        chai.expect(infoWithClinic).to.deep.nested.include({ 'transitions.registration.ok': true });
+      })
+      .then(() => utils.getDocs([patient._id, clinic._id]))
+      .then(([updWithPatient, updWithClinic]) => {
+        chai.expect(updWithPatient.scheduled_tasks).to.be.ok;
+        chai.expect(updWithPatient.scheduled_tasks).to.have.lengthOf(5);
+
+        chai.expect(updWithPatient.scheduled_tasks[0]).to.deep.nested.include(expectedMessage2('scheduled'));
+        chai.expect(updWithPatient.scheduled_tasks[1]).to.deep.nested.include(expectedMessage3('scheduled'));
+        chai.expect(updWithPatient.scheduled_tasks[2]).to.deep.nested.include(expectedMessage4('scheduled'));
+        chai.expect(updWithPatient.scheduled_tasks[3]).to.deep.nested.include(expectedMessage5('scheduled'));
+        chai.expect(updWithPatient.scheduled_tasks[4]).to.deep.nested.include(expectedMessage6('scheduled'));
+
+        // ensuring schedule started with our field and not reported_date
+        chai.expect(startDate.diff(updWithPatient.scheduled_tasks[3].due, 'days')).to.equal(-10);
+
+        chai.expect(updWithClinic.scheduled_tasks).to.be.ok;
+        chai.expect(updWithClinic.scheduled_tasks).to.have.lengthOf(3);
+
+        chai.expect(updWithClinic.scheduled_tasks[0]).to.deep.nested.include(expectedMessage2('scheduled'));
+        chai.expect(updWithClinic.scheduled_tasks[1]).to.deep.nested.include(expectedMessage3('scheduled'));
+        chai.expect(updWithClinic.scheduled_tasks[2]).to.deep.nested.include(expectedMessage4('scheduled'));
+      });
+  });
+});
```

Comment:
I think we'll need to assert on the due date as well, to make sure that the correct date is picked, not just that the message gets created.

Reply:
I've verified that by calculating difference between duedate and start_date and ensuring it matches our offset [here](https://github.com/medic/cht-core/pull/8639/files/aa30f4ed972750cb45adef30fe9a20f2d1edc6e5#diff-223091154c27112e2c6c6930b63c353c7ab2799567fb69f49546fffcccc5a384R243)

Reply:
Yea, I mean assert on every scheduled task, maybe included the expectedMessage function?